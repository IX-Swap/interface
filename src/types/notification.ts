import { AppFeature } from './app'

export type NotificationType =
  | 'success'
  | 'error'
  | 'info'
  | 'warning'
  | 'BUY'
  | 'SELL'

export interface Notification {
  _id: string
  createdBy: string
  createdAt: string
  to: string
  service: string
  feature: AppFeature
  resourceId: string
  subject: string
  type: NotificationType
  message: string
  read: boolean
  readAt: string
  comment?: string
}
