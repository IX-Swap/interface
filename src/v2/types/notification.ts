import { AppFeature } from './app'

export interface Notification {
  _id: string
  createdBy: string
  createdAt: string
  to: string
  service: string
  feature: AppFeature
  resourceId: string
  subject: string
  type: 'success' | 'failure'
  message: string
  read: boolean
  readAt: string
}
