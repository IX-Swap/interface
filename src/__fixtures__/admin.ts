import { user } from '__fixtures__/user'
import { UpdateUserRoleArgs } from 'v2/app/pages/admin/service/types'
import { AppRole } from 'v2/helpers/acl'

export const adminURLs = {
  setUserRole: `/auth/users/${user._id}/roles`
}

export const setUserRoleArgs: UpdateUserRoleArgs = {
  roles: [AppRole.ACCREDITED, AppRole.ADMIN].join(','),
  userId: user._id
}
