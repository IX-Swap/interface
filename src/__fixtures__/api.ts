import { APIServiceResponse } from 'v2/services/api/types'

export const url = 'http://example.com'

export const successfulResponse: APIServiceResponse<any> = {
  data: {},
  message: 'success',
  success: true
}

export const unsuccessfulResponse: APIServiceResponse = {
  data: undefined,
  message: 'error',
  success: false
}

export const headers = {
  Authorization: 'Bearer undefined'
}

export const customConfig = {
  headers: { 'Custom-Header': '12345' }
}

export const postJSON = { message: 'hello' }
export const postJSONString = JSON.stringify(postJSON)
export const postJSONHeaders = {
  ...headers,
  'Content-Type': 'application/json'
}

export const postFormData = new FormData()
export const postFormDataHeaders = headers
