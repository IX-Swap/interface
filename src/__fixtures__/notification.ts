import { AppFeature } from 'v2/types/app'
import { Notification } from 'v2/types/notification'

export const notification: Notification = {
  _id: '1',
  createdBy: '',
  createdAt: '2020-10-14T03:54:06.977Z',
  to: '',
  service: '',
  feature: AppFeature.Authentication,
  resourceId: '',
  subject: 'Cash Withdrawl',
  type: 'success',
  message: 'Your cash withdrawl request has been declined.',
  read: false,
  readAt: ''
}
