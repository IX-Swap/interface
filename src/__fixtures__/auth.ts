import { user } from './user'
import { APIServiceResponse } from 'v2/services/api/types'
import User from 'v2/types/user'
import {
  CompletePasswordResetArgs,
  LoginArgs,
  RequestPasswordResetArgs,
  SignupArgs,
  VerifySignupArgs
} from 'v2/types/auth'

const email = 'alex@investax.io'
const password = 'Qwerty12345_'

export const authURLs = {
  login: '/auth/sign-in',
  signup: '/auth/registrations',
  verifySignup: '/auth/registrations/confirm',
  getUser: '/auth/profiles',
  requestPasswordReset: '/auth/password/reset/start',
  completePasswordReset: '/auth/password/reset/confirm'
}

export const loginArgs: LoginArgs = {
  otp: '123456',
  password,
  email
}

export const signupArgs: SignupArgs = {
  name: 'Alex Solovev',
  email,
  password
}

export const verifySignupArgs: VerifySignupArgs = {
  verificationToken: 'mytoken'
}

export const requestPasswordResetArgs: RequestPasswordResetArgs = {
  email
}

export const completePasswordResetArgs: CompletePasswordResetArgs = {
  email,
  resetToken: 'token',
  newPassword: password
}

export const loginResponseSuccess: APIServiceResponse<User> = {
  data: user,
  message: 'Welcome',
  success: true
}

export const loginResponseFailure: APIServiceResponse<User> = {
  message: 'error',
  success: false
}

export const signupResponseSuccess: APIServiceResponse = {
  message: 'A verification e-mail has been sent to your account',
  success: true
}

export const signupResponseFailure: APIServiceResponse = {
  message: 'Sorry but this email address is already taken',
  success: false
}

export const verifySignupResponseSuccess: APIServiceResponse = {
  message: 'Your account has been verified',
  success: true
}

export const verifySignupResponseFailure: APIServiceResponse = {
  message: 'The token has already been verified, has expired, or is invalid',
  success: false
}

export const getUserResponseSuccess: APIServiceResponse<User> = {
  data: user,
  message: 'OK',
  success: true
}

export const getUserResponseFailure: APIServiceResponse<User> = {
  message: 'Access denied',
  success: false
}

export const requestPasswordResetResponseSuccess: APIServiceResponse = {
  message: 'OK',
  success: true
}

export const requestPasswordResetResponseFailure: APIServiceResponse = {
  message: 'Error',
  success: false
}

export const completePasswordResetResponseSuccess: APIServiceResponse = {
  message: 'Success',
  success: true
}

export const completePasswordResetResponseFailure: APIServiceResponse = {
  message: 'Reset token is already used, expired, or invalid',
  success: false
}
