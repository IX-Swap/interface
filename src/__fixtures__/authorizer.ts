import { Bank } from 'types/bank'
import { APIServiceResponse } from 'services/api/types'
import { CashDeposit } from 'types/cashDeposit'
import { Asset } from 'types/asset'
import { CorporateIdentity, IndividualIdentity } from 'types/identity'
import { user } from '__fixtures__/user'
import { CashWithdrawal } from 'types/cashWithdrawal'
import { DSWithdrawal } from 'types/dsWithdrawal'
import { DigitalSecurityOffering, DSOInsight } from 'types/dso'
import { AuthorizationInfo } from 'types/authorizer'
import { Commitment } from 'types/commitment'
import { emptyFile } from '__fixtures__/file'
import { network } from './network'
import { withdrawalAddress } from './withdrawalAddress'

export const asset: Asset = {
  _id: '5f732c538a568b50914d8372',
  deleted: false,
  createdBy: '5f73240373d4ab4b15fc1b2e',
  symbol: 'SGD',
  name: 'Singapore Dollars',
  type: 'Currency',
  numberFormat: {
    currency: 'SGD'
  },
  createdAt: '2020-09-29T12:45:07.411Z',
  updatedAt: '2020-10-05T03:14:17.244Z',
  description: 'test asset description'
}

export const address = {
  city: 'Omsk',
  country: 'Russian Federation',
  state: 'Siberia',
  line1: 'Address line 1',
  line2: 'Address line 2',
  postalCode: '123456'
}

export const authorizationInfo: AuthorizationInfo = {
  authorizer: 'id',
  comment: '',
  sharedWithUser: false,
  timestamp: ''
}

export const corporate: CorporateIdentity = {
  _id: '1',

  logo: '',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  documents: [],
  declarations: [],
  status: 'Submitted',
  beneficialOwners: [],
  companyAddress: address,
  companyLegalName: 'InvestaX',
  countryOfFormation: 'Singapore',
  dateOfIncorporation: '01-01-2000',
  directors: [],
  registrationNumber: '123456',
  representatives: [],
  email: '',
  contactNumber: '',
  user,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: []
}

export const individual: IndividualIdentity = {
  photo: '',
  _id: '1',
  email: 'email@example.com',
  annualIncome: '100000',
  bankAccountName: '',
  bankAccountNumber: '',
  bankName: '',
  contactNumber: '1234567890',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  countryOfResidence: 'Russian Federation',
  dob: 'DOB',
  employer: 'InvestaX',
  employmentStatus: 'Employed',
  firstName: 'John',
  lastName: 'Doe',
  gender: 'M',
  houseHoldIncome: '100000',
  industryOfEmployment: 'IT',
  maritalStatus: 'Married',
  middleName: '',
  nationality: 'Russian',
  occupation: 'Occupied',
  // politicallyExposed: false,
  sourceOfWealth: '___',
  status: 'Submitted',
  authorization: authorizationInfo,
  authorizationDocuments: [],
  authorizations: [],
  declarations: [],
  documents: [],
  address,
  user
}

export const dsWithdrawal: DSWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  recipientWallet: '0000000',
  asset,
  transaction: '',
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  }
}

export const dsoInsight: DSOInsight = {
  activityCount: 65,
  approvedcommitmentCount: 1,
  collectedOn: '2020-11-24T17:06:23.555Z',
  commitmentCount: 0,
  commitmentTotal: 0,
  investorCount: 1,
  raisedMax: 400,
  raisedMin: 400,
  raisedTotal: 400
}

export const dso: DigitalSecurityOffering = {
  _id: '1',
  asset: 'asset',
  businessModel: 'business model',
  capitalStructure: 'capital structure',
  network: network._id,
  corporate,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  createdBy: 'created by',
  currency: asset,
  deleted: false,
  deploymentInfo: undefined,
  distributionFrequency: 'distribution frequency',
  dividendYield: 1,
  documents: [],
  equityMultiple: 0,
  fundraisingMilestone: 'fundraising milestone',
  grossIRR: 0,
  interestRate: 1,
  introduction: 'introduction',
  investmentPeriod: 1,
  investmentStructure: 'investment structure',
  issuerName: 'issuer name',
  launchDate: '01-01-2000',
  leverage: 0,
  logo: '5f4f7d87f3e2c40bbab8a3f1',
  minimumInvestment: 100,
  pricePerUnit: 1,
  subscriptionDocument: emptyFile,
  team: [],
  tokenName: 'token name',
  tokenSymbol: 'token symbol',
  totalFundraisingAmount: 100000,
  useOfProceeds: 'use of proceeds',
  policyBuilder: undefined,
  user: user._id,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  },
  status: 'Approved',
  insight: dsoInsight
}

export const commitment: Commitment = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  status: 'Approved',
  currency: asset,
  createdBy: 'user',
  dso,
  identity: {
    individual,
    corporates: [corporate]
  },
  totalAmount: 0,
  authorizationDocuments: [],
  authorizationOverrides: [],
  authorizationOverride: undefined,
  authorization: undefined,
  authorizations: [],
  hold: 'dsadas',
  level: undefined,
  numberOfUnits: 0,
  pricePerUnit: 0,
  signedSubscriptionDocument: emptyFile,
  user,
  withdrawalAddress: withdrawalAddress
}

export const bank: Bank = {
  _id: '1',
  status: 'Approved',
  deleted: false,
  bankName: 'Rocketbank',
  bankAccountNumber: '1234567890',
  accountHolderName: 'Oleg Tinkoff',
  swiftCode: 'SWIFTCODE',
  authorized: true,
  authorizationDocuments: [],
  authorizations: [],
  supportingDocuments: [],
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  authorization: authorizationInfo,
  identity: {
    corporates: [corporate],
    individual
  },
  user,
  address,
  currency: asset
}

export const cashDeposit: CashDeposit = {
  _id: '1',
  amount: 10000,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  depositCode: 'DEPOSIT',
  status: 'Approved',
  user: '???',
  bankAccount: bank,
  individual,
  corporates: [corporate],
  asset,
  identity: {
    individual,
    corporates: [corporate]
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const cashWithdrawal: CashWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  corporates: [],
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  bank,
  individual,
  asset,
  identity: {
    individual,
    corporates: []
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const authorizerURLs = {
  approve: `/${bank._id}/approve`,
  reject: `/${bank._id}/reject`
}

export const approveResponseSuccess: APIServiceResponse = {
  message: 'Approved successfully',
  success: true
}

export const approveResponseFailure: APIServiceResponse = {
  message: 'Could not approve',
  success: false
}

export const rejectResponseSuccess: APIServiceResponse = {
  message: 'Rejected successfully',
  success: true
}

export const rejectResponseFailure: APIServiceResponse = {
  message: 'Could not reject',
  success: false
}
