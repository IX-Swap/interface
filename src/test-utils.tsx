import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
