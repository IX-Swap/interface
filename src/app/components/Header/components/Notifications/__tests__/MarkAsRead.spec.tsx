import React from 'react'
import { render } from 'test-utils'
import { notification } from '__fixtures__/notification'

import {
  MarkAsRead,
  MarkAsReadProps
} from 'app/components/Header/components/Notifications/MarkAsRead/MarkAsRead'

describe('MarkAsRead', () => {
  const props: MarkAsReadProps = {
    data: notification
  }

  afterAll(async () => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = render(<MarkAsRead {...props} />)
    expect(container).toMatchSnapshot()
  })
})
