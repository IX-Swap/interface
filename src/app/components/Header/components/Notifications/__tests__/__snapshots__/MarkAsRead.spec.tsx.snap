// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkAsRead should match snapshot 1`] = `
<div>
  <div
    aria-disabled="true"
    aria-label="Mark as read"
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root Mui-disabled makeStyles-container-1 makeStyles-container-2 css-slzo6x-MuiButtonBase-root-MuiButton-root"
    data-mui-internal-clone-element="true"
    role="button"
    tabindex="-1"
  />
</div>
`;
